FIX: wrong paths of installed headers in SofaBaseMechanics

marques-bruno opened the pull request:
I noticed that installed headers for SofaBaseMechanics aren't where they're supposed to be because the header's directory structure is not properly respected in SOFA's sofa_install_targets macro.

This commit fixes the issue for the specific case of SofaBaseMechanics/BarycentricMappers/*.{h,inl} but doesn't offer a definitive fix. Additionally, it duplicates headers in SofaBaseMechanics/ since the sofa_install_targets  is still in use.

A definitive solution would be to embed the fix in the sofa_install_targets macro, but I wanted your input first. @damienmarchal  @guparan  
 


______________________________________________________
<!--- Please leave this at the end of your message -->
This PR: 
- [ ] builds with SUCCESS for all platforms on the CI.
- [ ] does not generate new warnings.
- [ ] does not generate new unit test failures.
- [ ] does not generate new scene test failures.
- [ ] does not break API compatibility.
- [ ] is more than 1 week old (or has fast-merge label).

**Reviewers will merge only if all these checks are true.**

marques-bruno committed:
FIX: wrong paths to installed headers in SofaBaseMechanics

damienmarchal commented:
Isn't there a directory copy function in cmake ?

marques-bruno commented:
Well to my understanding it's supposed to be `install(DIRECTORY <dir> DESTINATION include/<dir>)`

Taken from [CMake's Doc](https://cmake.org/cmake/help/v3.0/command/install.html):
"The DIRECTORY form installs contents of one or more directories to a given destination. The directory structure is copied verbatim to the destination."

Obviously though, it's not working in this case. I can't figure out why not

marques-bruno committed:
FIX: remove PUBLIC_HEADER target properties

marques-bruno commented:
the install(DIRECTORY ... DESTINATION ...) was never called actually, since the pre-check 
`if("${public_header}" STREQUAL "public_header-NOTFOUND")` returned false because of the set_target_properties(PUBLIC_HEADER) declared before calling the sofa_install_targets macro...

marques-bruno commented:
This PR now fixes the issue cleanly. It would be interesting though, to figure out how many sofa libs are badly installed because of similar issues...

guparan commented:
Hey @marques-bruno welcome back in the PR business :+1: 
Did you try to only remove the set_target_properties(PUBLIC_HEADER) line? 
sofa_install_target should do the job.

marques-bruno committed:
CLEAN: removing the replacement code for proper header's install

marques-bruno commented:
Hey, thanks :)
Yeah, I figured it out...
I removed the workaround that I initially added and removed the PUBLIC_HEADER target property.
I think this PR is ready to be merged ;) :+1:

marques-bruno committed:
Update CMakeLists.txt

Removing meaningless comment
